File: android/app/src/main/java/com/lunacoding/stellar/firestore/Command.java
package com.lunacoding.stellar.firestore;

import com.google.firebase.Timestamp;
import com.google.firebase.firestore.QueryDocumentSnapshot;

import java.util.ArrayList;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

public class Command {
	public String command;
	public ArrayList<String> keyword;
	public String response;
	public boolean responseStandby;
	public Timestamp responseTimestamp;
	
	public Command() {
		this.command = "";
		this.keyword = new ArrayList<>();
		this.response = "";
		this.responseStandby = false;
		this.responseTimestamp = new Timestamp(new Date());
	}
	
	public Command(String command, ArrayList<String> keyword, String response, boolean responseStandby, Timestamp responseTimestamp) {
		this.command = command;
		this.keyword = keyword;
		this.response = response;
		this.responseStandby = responseStandby;
		this.responseTimestamp = responseTimestamp;
	}
	
	public Command(QueryDocumentSnapshot document) {
		this.command = document.getString("command");
		this.keyword = (ArrayList<String>) document.get("keyword");
		this.response = document.getString("response");
		this.responseStandby = document.getBoolean("responseStandby");
		this.responseTimestamp = document.getTimestamp("responseTimestamp");
	}
	
	public Command(Map map) {
		command = (String) map.get("command");
		keyword = (ArrayList<String>) map.get("keyword");
		response = (String) map.get("response");
		responseStandby = (boolean) map.get("responseStandby");
		responseTimestamp = (Timestamp) map.get("responseTimestamp");
	}
	
	public Map<String, Object> toMap() {
		Map map = new HashMap();
		map.put("command", command);
		map.put("keyword", keyword);
		map.put("response", response);
		map.put("responseStandby", responseStandby);
		map.put("responseTimestamp", responseTimestamp);
		
		return map;
	}
	
	public String toString() {
		return toMap().toString();
	}
}

File: android/app/src/main/java/com/lunacoding/naiclova/firestore/Command.kt
package com.lunacoding.naiclova.firestore

import com.google.firebase.Timestamp
import com.google.firebase.firestore.QueryDocumentSnapshot
import java.util.*

class Command {
	var key = ""
	var command: String?
	var keyword: ArrayList<String>?
	var response: String?
	var responseStandby: Boolean
	var responseTimestamp: Timestamp?

	constructor() {
		command = ""
		keyword = ArrayList()
		response = ""
		responseStandby = false
		responseTimestamp = Timestamp(Date())
	}

	constructor(command: String?, keyword: ArrayList<String>?, response: String?, responseStandby: Boolean, responseTimestamp: Timestamp?) {
		this.command = command
		this.keyword = keyword
		this.response = response
		this.responseStandby = responseStandby
		this.responseTimestamp = responseTimestamp
	}

	constructor(document: QueryDocumentSnapshot) {
		command = document.getString("command")
		keyword = document["keyword"] as ArrayList<String>?
		response = document.getString("response")
		responseStandby = document.getBoolean("responseStandby")!!
		responseTimestamp = document.getTimestamp("responseTimestamp")
	}

	constructor(key: String, document: QueryDocumentSnapshot) {
		this.key = key
		command = document.getString("command")
		keyword = document["keyword"] as ArrayList<String>?
		response = document.getString("response")
		responseStandby = document.getBoolean("responseStandby") ?: false
		responseTimestamp = document.getTimestamp("responseTimestamp")
	}

	constructor(map: Map<*, *>) {
		command = map["command"] as String?
		keyword = map["keyword"] as ArrayList<String>?
		response = map["response"] as String?
		responseStandby = map["responseStandby"] as Boolean
		responseTimestamp = map["responseTimestamp"] as Timestamp?
	}

	private fun toMap(): Map<String, Any?> {
		return mapOf(
				"command" to command,
				"keyword" to keyword,
				"response" to response,
				"responseStandby" to responseStandby,
				"responseTimestamp" to responseTimestamp,
		)
	}

	override fun toString(): String {
		return toMap().toString()
	}
}